tools/pubsys/src/repo.rs

```diff
@@ -388,20 +391,38 @@ fn is_file_not_found_error(e: &tough::error::Error) -> bool {
 }
 
 /// Gets the corresponding `KeySource` according to the signing key config from Infra.toml
-fn get_signing_key_source(signing_key_config: &SigningKeyConfig) -> Box<dyn KeySource> {
+fn get_signing_key_source(signing_key_config: &SigningKeyConfig) -> Result<Box<dyn KeySource>> {
     match signing_key_config {
-        SigningKeyConfig::file { path } => Box::new(LocalKeySource { path: path.clone() }),
-        SigningKeyConfig::kms { key_id } => Box::new(KmsKeySource {
+        SigningKeyConfig::file { path } => Ok(Box::new(LocalKeySource { path: path.clone() })),
+        SigningKeyConfig::kms { key_id, config, .. } => Ok(Box::new(KmsKeySource {
             profile: None,
-            key_id: key_id.clone(),
-            client: None,
+            key_id: key_id
+                .clone()
+                .context(error::MissingConfig { missing: "key_id" })?,
+            client: if let Some(region) = config
+                .as_ref()
+                .context(error::MissingConfig {
+                    missing: "KMS config",
+                })?
```

Comment:
This is making it optional. If `available_keys` is empty/ non-existent in the config (which is what Infra.toml looks like now), we just use the default region (like before).

Reply:
Interestingly, when `config` is not specified, since it's a flattened struct, it'll always be `Some`:
From infra.toml:
```
[repo.default]
....
signing_keys = { kms = { key_id = "arn:aws:kms:us-west-2:blah" } }

```

results in:
```
[pubsys/src/repo.rs:395] signing_key_config = kms {
    key_id: Some(
        "arn:aws:kms:us-west-2:blah",
    ),
    config: Some(
        KMSKeyConfig {
            available_keys: {},
            key_alias: None,
            regions: [],
            key_stack_arns: {},
        },
    ),
}

```

Should still not rely on this behavior though.